Problem with login / logout

It seems there is a problem with login and logout functionalities. I can't log out and then log in again.

```python
>>> from iotile_cloud.api.connection import Api
>>> domain = XXXX
>>> email = YYYY
>>> password= ZZZZ
>>> api = Api(domain=domain)
>>> api.login(email=email, password=password)
[DEBUG] 2018-12-13 15:17:58,197 Starting new HTTP connection (1): XXXX
[DEBUG] 2018-12-13 15:17:58,460 XXXX "POST /api/v1/auth/login/ HTTP/1.1" 200 772
[DEBUG] 2018-12-13 15:17:58,461 Welcome @YYYY
True
>>> api.logout()
[DEBUG] 2018-12-13 15:18:06,855 Starting new HTTP connection (1): XXXX
[DEBUG] 2018-12-13 15:18:06,873 XXXX "POST /api/v1/auth/logout/ HTTP/1.1" 204 0
[DEBUG] 2018-12-13 15:18:06,874 Goodbye @YYYY
>>> api.login(email=email, password=password)
[DEBUG] 2018-12-13 15:18:14,735 Starting new HTTP connection (1): localhost:9000
[DEBUG] 2018-12-13 15:18:14,747 http://localhost:9000 "POST /api/v1/auth/login/ HTTP/1.1" 403 19
[ERROR] 2018-12-13 15:18:14,748 Login failed: 403 "Already Logged-in"
False
>>> 
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,7 +1,10 @@
+### v0.9.9 (2018-12-13)
+
+  * Fix login. It is now possible to log in after logout.
+
 ### v0.9.8 (2018-11-08)
 
   * mock hooks updated for OTA
-
 
 ### v0.9.7 (2018-10-29)
 

--- a/iotile_cloud/api/connection.py
+++ b/iotile_cloud/api/connection.py
@@ -139,9 +139,9 @@ class RestResource(object):
         return url
 
     def _get_header(self):
-        headers = DEFAULT_HEADERS
+        headers = DEFAULT_HEADERS.copy()
         if self._store['use_token']:
-            if not "token" in self._store:
+            if "token" not in self._store:
                 raise RestBaseException('No Token')
             authorization_str = '{0} {1}'.format(self._store['token_type'], self._store["token"])
             headers['Authorization'] = authorization_str
@@ -179,7 +179,7 @@ class RestResource(object):
 
         try:
             resp = requests.patch(
-                self.url(), data=payload, headers=self._get_header(),params=kwargs,  verify=self._store['verify']
+                self.url(), data=payload, headers=self._get_header(), params=kwargs, verify=self._store['verify']
             )
         except requests.exceptions.SSLError as err:
             raise HttpCouldNotVerifyServerError("Could not verify the server's SSL certificate", err)
@@ -299,7 +299,7 @@ class Api(object):
 
     def logout(self):
         url = '{0}/{1}'.format(self.base_url, 'auth/logout/')
-        headers = DEFAULT_HEADERS
+        headers = DEFAULT_HEADERS.copy()
         headers['Authorization'] = '{0} {1}'.format(self.token_type, self.token)
 
         try:
@@ -317,8 +317,8 @@ class Api(object):
     def refresh_token(self):
         """
         Refresh JWT token
-        
-        :return: True if token was refreshed. False otherwise 
+
+        :return: True if token was refreshed. False otherwise
         """
         assert self.token_type == DEFAULT_TOKEN_TYPE
         url = '{0}/{1}'.format(self.base_url, 'auth/api-jwt-refresh/')
@@ -341,7 +341,6 @@ class Api(object):
         logger.error('Token refresh failed: ' + str(r.status_code) + ' ' + r.content.decode())
         self.token = None
         return False
-
 
     def __getattr__(self, item):
         """

--- a/tests/test_api.py
+++ b/tests/test_api.py
@@ -46,7 +46,11 @@ class ApiTestCase(unittest.TestCase):
             'jwt': 'big-token',
             'username': 'user1'
         }
-        m.post('http://iotile.test/api/v1/auth/login/', text=json.dumps(payload))
+
+        # login works only if there is no Authorization header in the request
+        def match_request_headers(request):
+            return 'Authorization' not in request.headers
+        m.post('http://iotile.test/api/v1/auth/login/', additional_matcher=match_request_headers, text=json.dumps(payload))
         m.post('http://iotile.test/api/v1/auth/logout/', status_code=204)
 
         api = Api(domain='http://iotile.test')
@@ -56,6 +60,10 @@ class ApiTestCase(unittest.TestCase):
         api.logout()
         self.assertEqual(api.username, None)
         self.assertEqual(api.token, None)
+
+        # can log in again
+        ok = api.login(email='user1@test.com', password='pass')
+        self.assertTrue(ok)
 
     @requests_mock.Mocker()
     def test_token_refresh(self, m):
@@ -211,4 +219,3 @@ class ApiTestCase(unittest.TestCase):
         api = Api(domain='http://iotile.test')
         with self.assertRaises(HttpServerError):
             api.test.post(payload)
-

--- a/version.py
+++ b/version.py
@@ -1,1 +1,1 @@
-version = '0.9.8'
+version = '0.9.9'

